src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubectlBuildWrapper.java

```diff
@@ -240,7 +251,8 @@ public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item item, @QueryPa
                                     CredentialsMatchers.instanceOf(StandardUsernamePasswordCredentials.class),
                                     CredentialsMatchers.instanceOf(TokenProducer.class),
                                     CredentialsMatchers.instanceOf(StandardCertificateCredentials.class),
-                                    CredentialsMatchers.instanceOf(FileCredentials.class)
+                                    CredentialsMatchers.instanceOf(FileCredentials.class),
+                                    CredentialsMatchers.instanceOf(GoogleRobotCredentials.class)
```

Comment:
This is not how we want to work with credentials. AFAIK there should be no need to touch the `kubernetes` plugin at all. Rather, the Google-related plugins should be implementing `TokenProducer`. That is how a generic plugin can support authentication with various credential types, including from AWS etc. I would have to refresh my memory on the details though.

Reply:
So, there is kubernetes-credentials-plugin, that defines TokenProducer. Should then my code be moved there, or google-oauth-plugin should implement that interface itself?

Reply:
There are bunch of credentials types implemented here: https://github.com/jenkinsci/kubernetes-credentials-plugin/tree/master/src/main/java/org/jenkinsci/plugins/kubernetes/credentials
One of them is `TokenProducer` implementation for OpenShift: 
https://github.com/jenkinsci/kubernetes-credentials-plugin/blob/master/src/main/java/org/jenkinsci/plugins/kubernetes/credentials/OpenShiftBearerTokenCredentialImpl.java

Reply:
Not sure offhand, would need to spend time researching it. I had thought the official API here was via the `authentication-tokens` plugin, which allows for example `docker-commons` to pick up Registry credentials from multiple vendors, but now I see that `kubernetes-credentials` indeed defines its own interface, perhaps in ignorance of that system. @Vlatombe @stephenc opinions?

Reply:
I think we deprecated the OpenShift kind already, in #529. #268 is the older change. Asking for advice from @maxlaverse.

Reply:
authentication-tokens should be able to provide the mapping... you want to turn the credentials into something that knows how to add auth to the HTTP client library... that may mean providing implementations of that for all the supported credential types... but it also means that the support can be added by extension plugins rather than needing to modify the base all the time. See https://github.com/jenkinsci/gitea-plugin/blob/master/src/main/java/org/jenkinsci/plugin/gitea/authentication/GiteaAuthUserSource.java and https://github.com/jenkinsci/gitea-plugin/blob/d5aa571acd4fb9e5d2f9c39034b577b397980f0b/src/main/java/org/jenkinsci/plugin/gitea/GiteaSCMSource.java#L720 for an example